testsuite/legacy-testsuite/src/test/java/com/sonatype/nexus/testsuite/ldap/AbstractLdapIT.java

```diff
@@ -12,92 +12,143 @@
+import org.sonatype.nexus.integrationtests.TestContainer;
 import org.sonatype.security.realms.ldap.api.dto.LdapConnectionInfoDTO;
 import org.sonatype.security.realms.ldap.api.dto.LdapServerConfigurationDTO;
 import org.sonatype.security.realms.ldap.api.dto.LdapServerRequest;
 import org.sonatype.security.realms.ldap.api.dto.LdapUserAndGroupAuthConfigurationDTO;
 import org.sonatype.security.realms.ldap.api.dto.XStreamInitalizer;
-
-import org.sonatype.ldaptestsuite.LdapServer;
-import org.sonatype.nexus.integrationtests.AbstractNexusIntegrationTest;
-import org.sonatype.nexus.integrationtests.TestContainer;
+import org.sonatype.security.realms.ldap.client.Configuration;
+import org.sonatype.security.realms.ldap.client.Connection;
+import org.sonatype.security.realms.ldap.client.Connection.Host;
+import org.sonatype.security.realms.ldap.client.Connection.Protocol;
+import org.sonatype.security.realms.ldap.client.LdapConfigurations;
+import org.sonatype.security.realms.ldap.client.Mapping;
 
 import com.google.common.base.Throwables;
 import com.thoughtworks.xstream.XStream;
 import org.apache.shiro.codec.Base64;
-import org.junit.AfterClass;
-import org.junit.Assert;
+import org.junit.After;
+import org.junit.Before;
 import org.junit.BeforeClass;
 import org.restlet.data.Response;
 
 import static org.apache.shiro.codec.CodecSupport.PREFERRED_ENCODING;
+import static org.sonatype.nexus.client.rest.BaseUrl.baseUrlFrom;
 
 public abstract class AbstractLdapIT
     extends AbstractNexusIntegrationTest
 {
-  public final String LDIF_DIR = "../../ldif_dir";
-
-  private static LdapServer ldapServer;
+  private LdapServer ldapServer;
 
   @BeforeClass
   public static void setSecurity() {
     TestContainer.getInstance().getTestContext().setSecureTest(true);
   }
 
-  @Override
-  protected void copyConfigFiles() throws IOException {
-    super.copyConfigFiles();
-
-    HashMap<String, String> map = new HashMap<String, String>();
-
-    // at this point the ldapServer should be running (if we have one)
-    if (this.isStartServer()) {
-      this.copyConfigFile("test.ldif", LDIF_DIR);
-
-      try {
-        ldapServer = TestContainer.getInstance().getPlexusContainer().lookup(LdapServer.class);
-        if (!ldapServer.isStarted()) {
-          ldapServer.start();
-        }
-      }
-      catch (Exception e) {
-        e.printStackTrace();
-        Assert.fail("Could not start or lookup ldap server ");
-      }
-
-      map.put("port", Integer.toString(ldapServer.getPort()));
-    }
-    else {
-      map.put("port", Integer.toString(12345));
+  @Before
+  public void beforeLdapTests() throws Exception {
+    ldapServer = new LdapServer(ldapServerConfiguration());
+    if (isStartLdapServer()) {
+      ldapServer.start();
     }
+    safeCreateLdapClientConfiguration();
+  }
 
-    // copy ldap.xml to work dir
-    this.copyConfigFile("ldap.xml", map, WORK_CONF_DIR);
+  protected boolean isStartLdapServer() {
```

Comment:
comment here might help to describe purpose and/or a potential rename: shouldStartLdapServer ?